src/main/java/com/flightstats/hub/dao/aws/ClusterContentService.java

```diff
@@ -134,16 +139,43 @@ public boolean historicalInsert(String channelName, Content content) throws Exce
         if (channel.isSingle()) {
             content = s3SingleContentDao.get(channelName, key);
         } else if (channel.isBatch()) {
-            content = s3BatchContentDao.get(channelName, key);
+            content = spokeReadContentDao.get(channelName, key);
+            if (content == null) {
+                content = getFromS3BatchAndWriteToSpokeBatch(channelName, key);
+            }
         } else {
-            content = s3SingleContentDao.get(channelName, key);
+            content = spokeReadContentDao.get(channelName, key);
+            if (content == null) {
+                content = s3SingleContentDao.get(channelName, key);
+            }
             if (content == null) {
-                content = s3BatchContentDao.get(channelName, key);
+                content = getFromS3BatchAndWriteToSpokeBatch(channelName, key);
             }
         }
         return checkForLargeIndex(channelName, content);
     }
 
+    private Content getFromS3BatchAndWriteToSpokeBatch(String channelName, ContentKey key) {
```

Comment:
maybe rename to storeS3BatchInSpokeReadCache?

Reply:
Ooo.. good catch.